knode/knode.go

```diff
@@ -0,0 +1,68 @@
+package knode
+
+import (
+	"errors"
+	"os"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var Kubeconfig *string
+
+type K8sNotifier struct {
+	nodeName string
+	config   *rest.Config
+	recorder record.EventRecorder
+}
+
+func NewK8sNotifier() *K8sNotifier {
+	return &K8sNotifier{}
+}
+
+// InitializeK8sNotifier sets up the event recorder to send events to the Kube API
+// server.
+func (k *K8sNotifier) InitializeK8sNotifier() error {
+	var err error
+	if k.config, err = config.GetConfig(); err != nil {
+		klog.Fatalf("cannot retrieve rest config for kubernetes. %v ", err)
+	}
+	// Proceed to fetch the NODE_NAME set through the downward API.
+	k.nodeName = os.Getenv("NODE_NAME")
+	// If the node name is still unset, obtain the hostname as it may be running in standalone
+	// mode.
+	if k.nodeName == "" {
+		k.nodeName, err = os.Hostname()
```

Comment:
may be we can make it as mandatory to set NODE_NAME environmental variable and avoid fetching hostname what if user dont know how node name is fetched and tries running from outside cluster

Reply:
By design in standalone mode, the application must run on the node from which the servicelogs needs to be processed + the dependency from Kubecofig.. In my opinion, obtaining the node name from the system removes chances where:
 1. The node with the set NODE_NAME environment variable may not exist. 
 2. Misconfiguration. 
 3. Typos(?). 

It is most likely that the hostname is not different from the kubernetes node-name.. Or maybe I'm missing the point why it'd be better to do away with the part where we determine hostname..